stl/inc/tuple

```diff
@@ -888,19 +1109,52 @@ _NODISCARD constexpr _Ty make_from_tuple(_Tuple&& _Tpl) { // construct _Ty from
 template <class... _Types, class _Alloc>
 struct uses_allocator<tuple<_Types...>, _Alloc> : true_type {}; // true_type if container allocator enabled
 
-#ifdef __cpp_lib_concepts
-template <class... _TTypes, class... _UTypes, template <class> class _TQual, template <class> class _UQual>
-    requires requires { typename tuple<common_reference_t<_TQual<_TTypes>, _UQual<_UTypes>>...>; }
-struct basic_common_reference<tuple<_TTypes...>, tuple<_UTypes...>, _TQual, _UQual> {
-    using type = tuple<common_reference_t<_TQual<_TTypes>, _UQual<_UTypes>>...>;
+#if _HAS_CXX23 && defined(__cpp_lib_concepts)
+template <_Tuple_like _TTuple, _Tuple_like _UTuple, template <class> class _TQual, template <class> class _UQual,
+    class _Indices = make_index_sequence<tuple_size_v<_TTuple>>>
+struct _Tuple_like_common_reference;
+
+template <class _TTuple, class _UTuple, template <class> class _TQual, template <class> class _UQual,
+    size_t... _Indices>
+    requires requires {
+                 typename tuple<common_reference_t<_TQual<tuple_element_t<_Indices, _TTuple>>,
+                     _UQual<tuple_element_t<_Indices, _UTuple>>>...>;

```

Comment:
I found a regressive bug of MSVC and reported it as DevCom-10265237. Should we add workaround here (ditto for `_Tuple_like_common_type`)?

Reply:
Specializations of `basic_common_reference` and `common_type` already check if tuples have the same size, so workaround is unnecessary here.